tests/fixture/testnet/local/node.go

```diff
@@ -0,0 +1,295 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package local
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cast"
+
+	"github.com/ava-labs/avalanchego/api/health"
+	cfg "github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/node"
+	"github.com/ava-labs/avalanchego/tests/fixture/testnet"
+	"github.com/ava-labs/avalanchego/utils/perms"
+)
+
+// Defines local-specific node configuration. Supports setting default
+// and node-specific values.
+//
+// TODO(marun) Support persisting this configuration per-node when
+// node restart is implemented. Currently it can be supplied for node
+// start but won't survive restart.
+type LocalConfig struct {
+	// Path to avalanchego binary
+	ExecPath string
+	// Whether a node is configured with static (e.g. 9560) or dynamic
+	// (e.g. 0) tcp ports for both staking and api connections.
+	UseStaticPorts bool
+	// The port number to start from when assigning static ports.
+	InitialStaticPort uint16
+}
+
+// Stores the configuration and process details of a node in a local network.
+type LocalNode struct {
+	testnet.NodeConfig
+	LocalConfig
+	node.NodeProcessContext
+
+	// Configuration is intended to be stored at the path identified in NodeConfig.Flags[cfg.DataDirKey]
+}
+
+func NewLocalNode(dataDir string) *LocalNode {
+	return &LocalNode{
+		NodeConfig: testnet.NodeConfig{
+			Flags: testnet.FlagsMap{
+				cfg.DataDirKey: dataDir,
+			},
+		},
+	}
+}
+
+// Attempt to read configuration and process details for a local node
+// from the specified directory.
+func ReadNode(dataDir string) (*LocalNode, error) {
+	node := NewLocalNode(dataDir)
+	if _, err := os.Stat(node.GetConfigPath()); err != nil {
+		return nil, fmt.Errorf("failed to read local node config file: %w", err)
+	}
+	if err := node.ReadAll(); err != nil {
+		return nil, err
+	}
+	return node, nil
```

Comment:
I'd generally remove the unneeded `err` check... but up to you if you feel like it's less readable here.
```suggestion
	return node, node.ReadAll()
```